public/js/search.js

```diff
@@ -0,0 +1,50 @@
+$(function(config){
+  var algolia = algoliasearch(config.applicationId, config.apiKey);
+  var index = algolia.initIndex(config.indexName);
+
+  hideSearch();
+
+  $('.sidebar-navigation--search input').on('input', function() {
+    var query = $(this).val();
+    onQueryChange(query);
+  });
+
+  function onQueryChange(query) {
+    if (query.length) {
+      index.search(query, {"hitsPerPage": "10", "page": "0", "typoTolerance": "false"}, onResult);
+    } else {
+      hideSearch();
+    }
+  };
+
+  function onResult(err, data) {
+    if (err) {
+      console.error(err);
+    } else {
+      displayResults(data);
+    }
+  };
+
+  function displayResults(data) {
```

Comment:
Another small nitpick but maybe something like this will be easier to work with when you change the UI

```js
function displayResults(data) {
  showSearch();
  $('.search-results').html('');
  if (data.hits.length) {
    postHits(data.hits)
  }
}

function postHits(hits) {
  var $resultsEl = $('.search-results')
  hits.forEach(function(result) {
    $resultsEl.append(resultTemplate(result))
  }
}

function resultTemplate(result) {
  `<li>
    <a href="${result.url}">${result.title}</a>
    ${result.raw_html}
  </li>`
}
```